driver/src/main/java/org/neo4j/driver/internal/async/NetworkSession.java

```diff
@@ -342,21 +341,14 @@ private void handleNewBookmark(DatabaseBookmark databaseBookmark) {
         var bookmark = databaseBookmark.bookmark();
         if (bookmark != null) {
             var bookmarks = Set.of(bookmark);
-            String databaseName = databaseBookmark.databaseName();
-            if (databaseName == null || databaseName.isEmpty()) {
-                databaseName = getDatabaseNameNow().orElse(FALLBACK_DATABASE_NAME);
-            }
             lastReceivedBookmarks = bookmarks;
-            bookmarkManager.updateBookmarks(databaseName, lastUsedBookmarks, bookmarks);
+            bookmarkManager.updateBookmarks(lastUsedBookmarks, bookmarks);
         }
     }
 
-    private Set<Bookmark> determineBookmarks(boolean useSystemOnly) {
-        var bookmarks = new HashSet<Bookmark>();
-        if (useSystemOnly) {
-            bookmarks.addAll(bookmarkManager.getBookmarks(DatabaseNameUtil.SYSTEM_DATABASE_NAME));
-        } else {
-            bookmarks.addAll(bookmarkManager.getAllBookmarks());
+    private Set<Bookmark> determineBookmarks(boolean dontUpdateLastUsedBookmarks) {
+        var bookmarks = new HashSet<>(bookmarkManager.getBookmarks());
+        if (!dontUpdateLastUsedBookmarks) {
```

Comment:
(Optional) This argument could be named something like `updateLastUsed`. Negation would become unnecessary.